### DISCUSSION TITLE
Font issues on Alpine

### DISCUSSION THREAD

**Discussion by 2fd** [samizdatco/skia-canvas#259] [Q&A] [answered]
I'm trying to create an image inside a node container on alpine, but the prop `front` doesn't update if I assign a new value.

As a Proof of Concept I setup this repo https://github.com/2fd/poc-skia-canvas-context to show the error.

It runs the following test:

```js
    // Create canvas  and context
    const canvas = new Canvas(100, 100)
    const context = canvas.getContext('2d')

    // Check default value
    expect(context.font).toBe('10px sans-serif')

    // Check update
    context.font = 'normal 600 20px/24px Times'
    expect(context.font).toBe('normal 600 20px/24px Times')
```

and run it on node `14`, `16`, `18`, and `20` using `slim` and `alpine` and it fails on every version of alpine

<img width="305" alt="Screenshot 2024-04-07 at 11 20 41 PM" src="https://github.com/samizdatco/skia-canvas/assets/208789/0b456fe1-7d32-4c2c-bd0f-2fe52edc1d51">

the value of font never was updated

<img width="510" alt="Screenshot 2023-01-23 at 01 15 49" src="https://user-images.githubusercontent.com/208789/213965032-bc5e12be-a25d-45c6-b868-1731adb3a83e.png">

**Comment by arthurbdev** [comment]
Did you find a workaround?

**Comment by 2fd** [comment]
~no, I didn't~

**Comment by 2fd** [comment]
Hi @arthurbdev, after digging a little bit a found the root cause [here](https://github.com/samizdatco/skia-canvas/blob/35ac526e8b428579f84b7d9557f032cbf5e04883/src/typography.rs#L496-L499). If you try to update the context with a font that is not present on your distro it will ignore any other change. In which case the solution will be [to install the font you will use](https://wiki.alpinelinux.org/wiki/Fonts). 

I have tested it and it worked just fine:
[<img width="309" alt="Screenshot 2024-04-07 at 11 50 11 PM" src="https://github.com/samizdatco/skia-canvas/assets/208789/62cf0db4-cee3-40a5-87fc-394415e29cf5">](https://github.com/2fd/poc-skia-canvas-context/actions/runs/8593853335)

**Comment by samizdatco** [comment, marked as answer]
By default, many Alpine-based docker images don't have **any** fonts installed, so it's important to either add something [to the system](https://wiki.alpinelinux.org/wiki/Fonts) or dynamically via [FontLibrary.use()](https://skia-canvas.org/api/font-library#use) if you want to be able to render text.
